### PR TITLE
Documentation for WebSocket Exporter

### PR DESCRIPTION
**Pull request by theMomax**
# Documentation for WebSocket Exporter

## :recycle: Current situation
No docs🤷‍♂️

## :bulb: Proposed solution
I wrote docs and adapted adapted the visions from the communicational patterns section to reflect the current state.

### Reviewer Nudging
Just read the docs...

### CODE REVIEW

File: Documentation/Communicational Patterns/2. Tooling/2.2. Parameter.md [hunk 1]
```diff
@@ -25,17 +25,17 @@ If the exporter allows for multiple messages to arrive on one instance of this e
 
 1. `Handler`s are not evaluated once for each `@Parameter` that is injected with a value, but once for each incoming message.
 2. Other events causing the `Handler` to be evaluated cannot be processed in parallel.
-3. A `DynamicProperty`'s `wrappedValue` cannot change while a `Handler` is evaluated except by the `Handler`'s `handle` function.
+3. Another `Property`'s public members cannot change while a `Handler` is evaluated except by the `Handler`'s `handle` function.
 
 ### Implementation Details
```
Comment (lschlesinger) [inline Documentation/Communicational Patterns/2. Tooling/2.2. Parameter.md]: nit: deviates from previously used "Implementation-Details"

File: Documentation/Communicational Patterns/3. Pattern Implementation/3.4. Service-Side Stream.md [hunk 2]
```diff
@@ -24,15 +24,15 @@ struct TimerObservable: ObservableObject {
...
+    @ObservedObject var timer = TimerObservable()
 
     func handle() -> String {
         return "Ping: \(name)"
     }
 }
 ```
 
-ObservableObjects that are exposed in the Environment are automatically observed.
+ObservableObjects can be globally exposed using keypaths.
```
Comment (lschlesinger) [inline Documentation/Communicational Patterns/3. Pattern Implementation/3.4. Service-Side Stream.md]: `ObservableObject`s ?

File: Documentation/Communicational Patterns/3. Pattern Implementation/3.1. Pattern Implementation.md [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+
+|           | Request-Response | Client-Side Stream | Service-Side Stream | Bidirectional Stream |
+|-----------|------------------|--------------------|---------------------|----------------------|
+| RESTful   | ✅                | ❌                  | ❌                   | ❌                    |
+| gRPC      | ✅                | ✅                  | ❌                   | ❌                    |
+| WebSocket | ✅                | ✅                  | ✅                   | ✅                    |
+| GraphQL   | ✅                | ❌                  | ❌                   | ❌                    |
+
 
 The following sections detail how different communicational patterns can be described using the tools described in the previous chapter.
```
Comment (lschlesinger) [inline Documentation/Communicational Patterns/3. Pattern Implementation/3.1. Pattern Implementation.md]: ... can be _implemented_ using the tools described in the previous chapter. 
Isn't it more implementation than description?

**Review by lschlesinger — APPROVED** [review]
Well done - very sophisticated documentation. As a consequence, I only found very nit-picking tiny details 😄 

Additionally, I have one remark regarding the structure. I found it somewhat irritating that _current_ and _future_ perspective (i.e., `documentation` and `vision` label) are mixed up within a single document. 
In other words, why is it called `document type`, if _one_ document can include multiple, nested `document type`s.
I observe, that you mostly explicitly use "future lingo" within `vision` subsections, but for example [here](https://github.com/Apodini/Apodini/blob/documentation/websocket-exporter-introduction/Documentation/Communicational%20Patterns/2.%20Tooling/2.1.%20Tooling.md) it is not immediately clear to me, that "Layers of Logic" is actually future vision. I know that the document label is indicating this, but imo the language does not.

File: Documentation/Communicational Patterns/3. Pattern Implementation/3.1. Pattern Implementation.md (hunk 3 above)
Comment (theMomax) [reply Documentation/Communicational Patterns/3. Pattern Implementation/3.1. Pattern Implementation.md]: Yeah, makes a lot more sense🙈

File: Documentation/Communicational Patterns/3. Pattern Implementation/3.4. Service-Side Stream.md (hunk 2 above)
Comment (theMomax) [reply Documentation/Communicational Patterns/3. Pattern Implementation/3.4. Service-Side Stream.md]: Thanks👍🏻

**Comment by theMomax** [comment]
> I observe, that you mostly explicitly use "future lingo" within `vision` subsections, but for example [here](https://github.com/Apodini/Apodini/blob/documentation/websocket-exporter-introduction/Documentation/Communicational%20Patterns/2.%20Tooling/2.1.%20Tooling.md) it is not immediately clear to me, that "Layers of Logic" is actually future vision.

I agree. I changed the type of this section to `concept` which I believe is more suiting. Thanks for pointing out all the small details that didn't catch my eye.

**Pull request merged**